`sr3 declare` fails because `tlsRigour` isn't found

## How to recreate
Run `sr3 --users declare my-component/my-config`

You should get the following errors when trying to declare a new user.

```
2026-03-16 16:41:42,076 3794485 [ERROR] sarracenia.moth.amqp putSetup failed to connect or declare exchanges ['xs_MY_NEW_USER']@MY_NEW_USER on fluxi.cmc.ec.gc.ca: 'tlsRigour'
2026-03-16 16:41:42,076 3794485 [DEBUG] sarracenia.moth.amqp putSetup Exception details:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/sarracenia/moth/amqp.py", line 465, in putSetup
    if not self.__connect(self.o['broker']):
  File "/usr/lib/python3/dist-packages/sarracenia/moth/amqp.py", line 236, in __connect
    if self.o['tlsRigour'] == 'lax':
KeyError: 'tlsRigour'

```

I was able to work around the problem by adding a `tlsRigour : 'normal'` inside of the default_options of the amqp MOTH class.

i.e. https://github.com/MetPX/sarracenia/blob/4fb8a33041e0c6530e4c7b5f10ed7ab9b43bb28a/sarracenia/moth/amqp.py#L53-L70

I think adding that default `tlsRigour` value to all of the MOTH types might be a good enough fix.

I just realized that this bug also affects `sr3 cleanup` and will **prevent the removal of the queue from the broker** when the `broker` entry in the configuration starts with `amqps`.  I've tested on multiple environments (U18.04/U22.04) and multiple clusters (ddsr-dev,ddi) and the result is the same.
```
sr3 --debug cleanup poll/edcm-poes-dev1
2026-03-18 21:22:43,846 46023 [ERROR] sarracenia.moth.amqp getSetup connecting to: q_MSC-SAT-DEV.poll.edcm-poes-dev1.ddsr-dev01, durable: True, expire: 21600.0, auto_delete=False
2026-03-18 21:22:43,846 46023 [ERROR] sarracenia.moth.amqp getSetup failed connection to amqps://MSC-SAT-DEV@my-broker/: 'tlsRigour'
 2026-03-18 21:22:43,846 46023 [DEBUG] sarracenia.moth.amqp getSetup Exception details:
Traceback (most recent call last):
File "/local/home/sarra/sr3/sarracenia/moth/amqp.py", line 384, in getSetup
if not self.__connect(broker):
File "/local/home/sarra/sr3/sarracenia/moth/amqp.py", line 234, in __connect
if self.o['tlsRigour'] == 'lax':
KeyError: 'tlsRigour'
```

--- a/sarracenia/config/publisher.py
+++ b/sarracenia/config/publisher.py
@@ -53,6 +53,9 @@ class Publisher(dict):
             logger.error("malformed publisher, missing (post_)exchange")
             return
 
+        if hasattr(options,'tlsRigour') :
+            self['tlsRigour'] = options.tlsRigour
+    
         if hasattr(options,'post_format') :
             self['format'] = options.post_format
         elif hasattr(options,'post_topicPrefix') and options.post_topicPrefix[0] in [ 'v02', 'v03' ]:

--- a/sarracenia/moth/amqp.py
+++ b/sarracenia/moth/amqp.py
@@ -184,7 +184,7 @@ class AMQP(Moth):
 
         # update self.o (already set by super().__init__) with AMQP-specific defaults,
         # then re-apply props so they take priority.
-        self.o.update(copy.deepcopy(default_options))
+        self.o.update(default_options)
         self.o.update(props)
 
         self.first_setup = True

--- a/sarracenia/moth/amqp.py
+++ b/sarracenia/moth/amqp.py
@@ -182,7 +182,9 @@ class AMQP(Moth):
             format=
             '%(asctime)s [%(levelname)s] %(name)s %(funcName)s %(message)s')
 
-        self.o = copy.deepcopy(default_options)
+        # update self.o (already set by super().__init__) with AMQP-specific defaults,
+        # then re-apply props so they take priority.
+        self.o.update(copy.deepcopy(default_options))
         self.o.update(props)
 
         self.first_setup = True

--- a/sarracenia/sr.py
+++ b/sarracenia/sr.py
@@ -1848,7 +1848,8 @@ class sr_GlobalState:
                             'echangeDeclare': False,
                             'subscription_index': 0,
                             'subscriptions' : [ s ],
-                            'message_strategy': { 'stubborn':True }
+                            'message_strategy': { 'stubborn':True },
+                            'tlsRigour': q['tlsRigour']
                         })
                     qdc.getSetup()
                     qdc.getCleanUp()

--- a/sarracenia/sr.py
+++ b/sarracenia/sr.py
@@ -1542,7 +1542,8 @@ class sr_GlobalState:
                         'broker': self.default_cfg.admin,
                         'dry_run': self.options.dry_run,
                         'exchange': self.default_cfg.declared_exchanges,
-                        'message_strategy': { 'stubborn':True }
+                        'message_strategy': { 'stubborn':True },
+                        'tlsRigour': self.options.tlsRigour
                     })
                 xdc.putSetup()
                 xdc.close()
@@ -1564,7 +1565,8 @@ class sr_GlobalState:
                                 'broker': p['broker'],
                                 'dry_run': self.options.dry_run,
                                 'exchange': p['exchange'],
-                                'message_strategy': { 'stubborn':True }
+                                'message_strategy': { 'stubborn':True },
+                                'tlsRigour': p['tlsRigour']
                             })
                          xdc.putSetup()
                          xdc.close()
@@ -1882,7 +1884,8 @@ class sr_GlobalState:
                                         'exchange': p['exchange'],
                                         'dry_run': self.options.dry_run,
                                         'broker': self.brokers[h]['admin'],
-                                        'message_strategy': { 'stubborn':True }
+                                        'message_strategy': { 'stubborn':True },
+                                        'tlsRigour': p['tlsRigour']
                                     })
                                 if qdc:
                                     qdc.putSetup()

--- a/tests/sarracenia/moth/amqp_test.py
+++ b/tests/sarracenia/moth/amqp_test.py
@@ -1,6 +1,56 @@
 import pytest
 from tests.conftest import *
-#from unittest.mock import Mock
+from unittest.mock import patch, MagicMock
 
 import sarracenia.config
+import sarracenia.moth
 import sarracenia.moth.amqp
+
+
+class Test_AMQPDefaultOptions:
+    """Test that AMQP inherits tlsRigour from parent Moth defaults.
+
+    Regression test for https://github.com/MetPX/sarracenia/issues/1591
+    When sr3 declare creates an AMQP instance with minimal props (no tlsRigour),
+    __connect would raise KeyError: 'tlsRigour'.
+    """
+
+    @patch('sarracenia.moth.amqp.amqp')
+    def test_tlsRigour_present_with_minimal_props(self, mock_amqp_lib):
+        """AMQP init with minimal props (like sr3 declare) must still have tlsRigour."""
+        minimal_props = {
+            'broker': MagicMock(),
+            'dry_run': False,
+            'exchange': 'xpublic',
+            'message_strategy': {'stubborn': True},
+        }
+        instance = sarracenia.moth.amqp.AMQP(minimal_props, is_subscriber=False)
+        assert 'tlsRigour' in instance.o
+        assert instance.o['tlsRigour'] == 'normal'
+
+    @patch('sarracenia.moth.amqp.amqp')
+    def test_tlsRigour_override_preserved(self, mock_amqp_lib):
+        """When props explicitly set tlsRigour, the value must be kept."""
+        props = {
+            'broker': MagicMock(),
+            'dry_run': False,
+            'exchange': 'xpublic',
+            'message_strategy': {'stubborn': True},
+            'tlsRigour': 'lax',
+        }
+        instance = sarracenia.moth.amqp.AMQP(props, is_subscriber=False)
+        assert instance.o['tlsRigour'] == 'lax'
+
+    @patch('sarracenia.moth.amqp.amqp')
+    def test_amqp_specific_defaults_still_applied(self, mock_amqp_lib):
+        """AMQP-specific defaults (e.g. vhost, exchangeDeclare) must still be set."""
+        minimal_props = {
+            'broker': MagicMock(),
+            'dry_run': False,
+            'exchange': 'xpublic',
+            'message_strategy': {'stubborn': True},
+        }
+        instance = sarracenia.moth.amqp.AMQP(minimal_props, is_subscriber=False)
+        assert instance.o['vhost'] == '/'
+        assert instance.o['exchangeDeclare'] is True
+        assert instance.o['durable'] is True

